### PR TITLE
Update `pre-commit` configuration to check file changes

### PR DESCRIPTION
**Pull request by yusufsheiqh**
## Context

- Update `pre-commit` configuration to check for tracked file changes as a result of `pre-commit` hooks
  - Stash all changes, check if there are any changes, remove stashed state.
- Update order of `pre-commit` hooks
  - Remove stashed state at the end of the `pre-commit` workflow.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,4 +1,12 @@
 repos:
+  - repo: local
+    hooks:
+      - id: stash-all-files
```
Comment (lgarvey) [inline .pre-commit-config.yaml:4]: Ah, so only staged files will have precommit hook checks ran on them?  This probably fits my workflow better as I never do a `git add --all`  👍

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -65,3 +65,27 @@ repos:
...
+        pass_filenames: false
+        always_run: true
+  - repo: local
+    hooks:
+      - id: pop-stashed-files
+        name: Remove stash state and apply on current working tree state
+        entry: git stash pop
+        language: system
+        pass_filenames: false
+        always_run: true
```
Comment (WillGibson) [inline .pre-commit-config.yaml:91]: Does this mean that it will always run even if an earlier hook failed?

Comment (WillGibson) [reply .pre-commit-config.yaml:91]: Yes, as discussed verbally just now.

**Review by WillGibson — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (yusufsheiqh) [reply .pre-commit-config.yaml:4]: Correct! 😄

**Pull request merged**